libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/MapboxSpeechPlayer.java

```diff
@@ -129,12 +139,12 @@ private void stopMediaPlayerPlaying() {
       if (mediaPlayer != null && mediaPlayer.isPlaying()) {
         mediaPlayer.stop();
         mediaPlayer.release();
-        if (instructionListener != null) {
-          instructionListener.onDone();
+        if (speechListener != null) {
```

Comment:
you shouldn't need this null check anymore since `SpeechListener` is a constructor parameter now.